parsing/unit_info.mli

```diff
@@ -0,0 +1,136 @@
+
+(** [modname_from_source filename] is [modulize stem] where [stem] is the
+    basename of the filename [file] stripped from all its extensions.*)
+val modname_from_source: string -> string
+
+(** {2:module_name_validation Module name validation function}*)
+
+(** [is_unit_name ~strict name] returns true only if [name] can be used as a
+    correct module name. The [strict] mode enforces that the first character is
+    an ascii capital letter. *)
+val is_unit_name : strict:bool -> string -> bool
+
+(** [check_unit_name file] prints a warning if the derived module name [modname
+    file] should not be used as a module name. *)
+val check_unit_name : (string,_) t -> unit
+
+(** {1:info_creation Creation of file metadata} *)
+
+(** [from_source filename] associates the module name [modname_from_source
+    filename] to the source file [filename]. *)
+val from_source: string -> source_file
+
+(** [from_target filename] associates the target [filename] to the module name
+    [modname_from_source filename]. *)
+val from_target: string -> target_only
+
+(** [from_target_prefix ~source_file target_prefix] aasociates both the
+    [source_file] and the module name [modname_from_source target_prefix] to
+    [target_prefix].
+
+   This function emits a warning if the derived module name is not valid
+   according to {!is_unit_name}[ ~strict:true].
+*)
+val from_target_prefix: source_file:string -> string -> target_prefix
+
+(** {1:info_build_artifacts Build artifacts} *)
+
+(** Those functions changes the target [filename] while conserving both source
+    file and module name metadata.*)
+val cmi: ('a,_) t -> 'a any_target
+val cmt: ('a,_) t -> 'a any_target
+val cmti: ('a,_) t -> 'a any_target
+val annot: ('a,_) t -> 'a any_target
+val obj: ('a,_) t -> 'a any_target
+
+(** The functions below return the filename of build artefact associated to a
+    file. *)
+val cmx_name: _ t -> string
+val cmo_name: _ t -> string
+val obj_name: _ t -> string
+val annot_name: _ t -> string
+
+(** {1:ml_mli_cmi_interaction Mli and cmi files derived from implementation
+    files } *)
+
+(** The compilation of module implementation changes in presence of mli and cmi
+    files, the function belows help to handle this. *)
+
+(** [mli_from_source u] is the interface source filename associated to the
+    implementation source file [u]. The actual suffix depends on
+    {!Config.interface_suffix}.
+*)
+val mli_from_source: source_file -> string
+
+(** [mli_from_target t] is the name of the interface source file derived from
+    the compilation target [t]. This variant is necessary when building
+    implementation targets which do not have an associated source file (e.g.
+    packed modules). *)
+val mli_from_target: target_only -> string
+
+(** Check if the file is a cmi *)
+val is_cmi: _ any_target -> bool
+
+(** [cmi_uncap u] finds in the load_path a name matching the module name
+    [modname u]. *)
+val cmi_uncap: _ t -> target_only
```

Comment:
I would name this `find_cmi_uncap` to keep in mind that there is a non-trivial lookup logic behind it (with a performance cost, etc.).